src/components/claiming/Loading.vue

```diff
@@ -32,13 +32,17 @@ export default {
   computed: {
     ...mapState({
       sessionId: state => state.wallet.sessionId,
+      disclaimers: state => state.wallet.disclaimers,
       error: state => state.wallet.errors.unlockWallet || state.wallet.errors.createWallet,
     }),
   },
   watch: {
     async sessionId(value) {
       if (value) {
         await sleep(2000)
+        Object.entries(this.disclaimers).forEach(elm => {
+          this.$store.dispatch('signData', { data: elm[0] })
```

Comment:
Instead of dispatch multiple times signData, I would dispatch it only once and send multiple sign data request from que vuex action.